packages/mdx/index.js

```diff
@@ -30,6 +30,7 @@ function createMdxAstCompiler(options) {
   const fn = unified()
     .use(toMDAST, options)
     .use(remarkMdx, options)
+    .use(mdxHastToJsx, options) // Set JSX compiler early so it can be overridden
```

Comment:
Looks like this is causing some issues in other tests. Wonder if we should instead use (and document) the [`compilers` API](https://github.com/mdx-js/mdx/blob/master/packages/mdx/index.js#L62)?